File: DSA/Devansh-Nishchal-Mandloi/day16que1.cpp
#include <iostream>
#include <vector>
using namespace std;

// Definition for a binary tree node.
struct TreeNode {
    int val;
    TreeNode* left;
    TreeNode* right;
    TreeNode(int x) : val(x), left(NULL), right(NULL) {}
};

// Function to perform postorder traversal.
void postorderTraversal(TreeNode* root, vector<int>& result) {
    if (root == NULL) {
        return;
    }
    postorderTraversal(root->left, result);
    postorderTraversal(root->right, result);
    result.push_back(root->val);
}

// Example usage
int main() {
    // Creating a simple binary tree
    TreeNode* root = new TreeNode(1);
    root->left = new TreeNode(2);
    root->right = new TreeNode(3);
    root->left->left = new TreeNode(4);
    root->left->right = new TreeNode(5);

    vector<int> result;
    postorderTraversal(root, result);
    for (int val : result) {
        cout << val << " ";
    }
    cout << endl;

    return 0;
}

File: DSA/Priyanshi/Day16_que1.java
class Node {
	int data;
	Node left, right;
	Node(int v)
	{
		data = v;
		left = right = null;
	}
}

class Day16_que1 {

	static void printPostorder(Node node)
	{
		if (node == null)
			return;

		printPostorder(node.left);

		printPostorder(node.right);

		System.out.print(node.data + " ");
	}

	public static void main(String[] args)
	{
		Node root = new Node(1);
		root.left = new Node(2);
		root.right = new Node(3);
		root.left.left = new Node(4);
		root.left.right = new Node(5);

		System.out.println("Postorder traversal of binary tree is: ");
		printPostorder(root);
	}
}
